mocks/log.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2015 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package mocks
+
+import (
+	"log"
+	"log/syslog"
+	"regexp"
+
+	blog "github.com/letsencrypt/boulder/log"
+)
+
+// MockSyslogWriter implements the blog.SyslogWriter interface. It
+// stores all logged messages in a buffer for inspection by test
+// functions (via GetAll()) instead of sending them to syslog.
+type MockSyslogWriter struct {
+	logged    []*LogMessage
+	msgChan   chan<- *LogMessage
+	getChan   <-chan []*LogMessage
+	clearChan chan<- struct{}
+	closeChan chan<- struct{}
+}
+
+// LogMessage is a log entry that has been sent to a MockSyslogWriter.
+type LogMessage struct {
+	Priority syslog.Priority // aka Log level
+	Message  string          // content of log message
+}
+
+var levelName [8]string
+
+func init() {
```

Comment:
Fair point. I liked the efficiency of an array, but the value (compared to simplicity of source code) is dubious in a test suite, isn't it.